### PR TITLE
chore(deps): Bump base-controller to ^5.0.1

### PR DESCRIPTION
**Pull request by mcmire**
## **Description**

This version of `@metamask/base-controller` does not introduce any functional changes, but there are a few developer experience improvements:

- The controller messenger is now more type-safe, in that a messenger with an empty action/event allowlist guarantees that no actions/events are allowed to be called/emitted, and conversely, that a non-empty allowlist only allows the given actions/events
- The "selector" version of `subscribe` (three arguments instead of two) is now fixed so that the arguments to the selector function are correctly typed.
- It is now impossible to mutate state directly; state must only be mutated inside of a controller using the `update` method. This ensures that state updates are always controlled and expected.

View changelog: https://github.com/MetaMask/core/blob/%40metamask/base-controller%405.0.1/packages/base-controller/CHANGELOG.md

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24209?quickstart=1)

## **Related issues**

Progresses #24140.

## **Manual testing steps**

(None needed as there should be no functional changes.)

## **Screenshots/Recordings**

(None)

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by mcmire** [comment]
@metamaskbot update-policies

**Comment by mikesposito** [comment]
@metamaskbot update-policies

**Comment by mikesposito** [comment]
@metamaskbot update-policies

**Review by cryptodev-2s — DISMISSED** [review]
LGTM!

**Comment by mikesposito** [comment]
@metamaskbot update-policies

**Review by legobeat — DISMISSED** [review]

**Comment by mcmire** [comment]
@metamaskbot update-policies

**Comment by mcmire** [comment]
@metamaskbot update-policies

**Review by legobeat — DISMISSED** [review]

File: app/scripts/controllers/authentication/authentication-controller.test.ts [hunk 1]
```diff
@@ -254,6 +254,7 @@ function createAuthenticationMessenger() {
   return messenger.getRestricted({
     name: 'AuthenticationController',
     allowedActions: [`SnapController:handleRequest`],
+    allowedEvents: [],
```
Comment (mcmire) [inline app/scripts/controllers/authentication/authentication-controller.test.ts:257]: Prior to `@metamask/base-controller` 5.0.0, `allowedEvents` was optional if no events were allowlisted; now this argument is required. See changelog: https://github.com/MetaMask/core/blob/main/packages/base-controller/CHANGELOG.md#changed-2

File: app/scripts/controllers/user-storage/user-storage-controller.test.ts [hunk 2]
```diff
@@ -301,6 +301,7 @@ function mockUserStorageMessenger() {
       'MetamaskNotificationsController:disableMetamaskNotifications',
       'MetamaskNotificationsController:selectIsMetamaskNotificationsEnabled',
     ],
+    allowedEvents: [],
```
Comment (mcmire) [inline app/scripts/controllers/user-storage/user-storage-controller.test.ts:304]: Prior to `@metamask/base-controller` 5.0.0, `allowedEvents` was optional if no events were allowlisted; now this argument is required. See changelog: https://github.com/MetaMask/core/blob/main/packages/base-controller/CHANGELOG.md#changed-2

File: app/scripts/lib/snap-keyring/utils/isBlockedUrl.test.ts [hunk 3]
```diff
@@ -6,11 +6,10 @@ describe('isBlockedUrl', () => {
   const messenger = new ControllerMessenger();
   const phishingControllerMessenger = messenger.getRestricted({
     name: 'PhishingController',
+    allowedActions: [],
```
Comment (mcmire) [inline app/scripts/lib/snap-keyring/utils/isBlockedUrl.test.ts:9]: Prior to `@metamask/base-controller` 5.0.0, `allowedActions` and `allowedEvents` were optional if no actions or events were allowlisted; now these arguments are required. See changelog: https://github.com/MetaMask/core/blob/main/packages/base-controller/CHANGELOG.md#changed-2

**Comment by Gudahtt** [comment]
Looks like e2e tests are failing

**Review by Prithpal-Sooriya — DISMISSED** [review]

**Comment by mikesposito** [comment]
@metamaskbot update-policies

**Comment by mikesposito** [comment]
@Gudahtt I don't think they are related to this PR specifically. They also became green after a re-run

**Comment by mikesposito** [comment]
@metamaskbot update-policies

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Comment by mcmire** [comment]
@metamaskbot update-policies

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Review by DDDDDanica — DISMISSED** [review]

**Comment by DDDDDanica** [comment]
LGTM !

**Review by Prithpal-Sooriya — DISMISSED** [review]

**Comment by mikesposito** [comment]
@metamaskbot update-policies

**Review by Gudahtt — DISMISSED** [review]

**Comment by Gudahtt** [comment]
@metamaskbot update-policies

**Review by Gudahtt — APPROVED** [review]

**Review by davidmurdoch — APPROVED** [review]

**Review by Prithpal-Sooriya — APPROVED** [review]

**Review by gantunesr — APPROVED** [review]

**Pull request merged**